modules/openapi-generator/src/main/resources/kotlin-multiplatform-client/common/main/infrastructure/ApiClientBase.kt.mustache

```diff
@@ -137,13 +135,13 @@ public open class ApiClientBase {
     }
 
     protected suspend fun jsonRequest(requestConfig: RequestConfig, body: Any? = null, authNames: List<String>): HttpResponse {
-        val contentType = (requestConfig.headers[HttpHeaders.ContentType]?.let { ContentType.parse(it) }
-                ?: ContentType.Application.Json)
-        return if (body != null) request(requestConfig, serializer.write(body, contentType), authNames)
+        if (HttpHeaders.ContentType !in requestConfig.headers)
+        requestConfig.headers[HttpHeaders.ContentType] = ContentType.Application.Json.toString()
+        return if (body != null) request(requestConfig, body, authNames)
         else request(requestConfig, authNames = authNames)
     }
 
-    protected suspend fun request(requestConfig: RequestConfig, body: OutgoingContent = EmptyContent, authNames: List<String>): HttpResponse {
+    protected suspend fun request(requestConfig: RequestConfig, body: Any = EmptyContent, authNames: List<String>): HttpResponse {
```

Comment:
Any is almost never a good idea

Reply:
Right but I'm not sure that serialising the request body instead of having Ktor do it(which it does well as long as models are generated correctly) and than filtering-out the ContentType header from the outgoing request  is the way to go..

Reply:
So the serialization of ktor still works in this... interesting implementation?

Reply:
If so you just need to update the branch and then I can merge it (plus optionally the common code for the http client creation)